AndelaOSP/shufflebox-backend
# Ft models hotfix

#### What does this PR do?
* It refactors brownbag model and renames all things `BrownBag` to `Brownbag`
* It defines a way to serialize brownbag so that the data returned is of the following format:
```
{
  "id": 1,
  "date": "2017-04-07",
  "status": "next_in_line",
  "user": {
    "id": 3,
    "username": "some_user",
    "email": "some_user@gmail.com",
    "profile": {
      "avatar": "avatar.png",
      "birth_date": null,
      "bio": "I have a dream"
    }
  }
}
```
* It disables direct creation of Brownbag so that you can only use `api/shuffle/` endpoint to do so. Getting a specific Brownbag and updating can still be performed under `/api/brownbags/<id>`
* It limits a Brownbag to once a every Friday. If one creates another brownbag after creating one, it will be pushed to the next Friday's date.
* Removes the unnecessary use of limit in Brownbag creation

#### How should this be manually tested?
* After pulling for changes, delete the old dev db, make and re-run migrations.
* Access `api/users/` and create enough users to test
* With a rest client eg Postman, issue a POST request to the endpoint `api/shuffle/` with the following JSON data: (PS -> Ensure you have a JWT token on the Authorization header before making the request.)
 ` { "type": "brownbag" }`
 Issuing the same request again creates a new Brownbag and slots it for the next Friday after the already taken Friday date.
